{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Current working dir: /Users/jakeharris/Documents/GitHub/df-assignment\n"
     ]
    }
   ],
   "source": [
    "# Packages\n",
    "import os\n",
    "import sys\n",
    "import math\n",
    "import re\n",
    "import scipy\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Check working directory\n",
    "print('Current working dir:', os.getcwd())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Patient_ID', 'Tumor_Sample_Barcode', 'Matched_Norm_Sample_Barcode',\n",
      "       'Response', 'Silent_mutations_per_Mb', 'Nonsynonymous_mutations_per_Mb',\n",
      "       'Mutations_per_Mb'],\n",
      "      dtype='object')\n",
      "Number of patient entries: 50\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Matched_Norm_Sample_Barcode</th>\n",
       "      <th>Response</th>\n",
       "      <th>Silent_mutations_per_Mb</th>\n",
       "      <th>Nonsynonymous_mutations_per_Mb</th>\n",
       "      <th>Mutations_per_Mb</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Patient-0</td>\n",
       "      <td>Patient-0-Tumor</td>\n",
       "      <td>Patient-0-Normal</td>\n",
       "      <td>Non-Responder</td>\n",
       "      <td>2.87</td>\n",
       "      <td>6.77</td>\n",
       "      <td>9.64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Patient-1</td>\n",
       "      <td>Patient-1-Tumor</td>\n",
       "      <td>Patient-1-Normal</td>\n",
       "      <td>Responder</td>\n",
       "      <td>1.92</td>\n",
       "      <td>6.14</td>\n",
       "      <td>8.06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Patient-2</td>\n",
       "      <td>Patient-2-Tumor</td>\n",
       "      <td>Patient-2-Normal</td>\n",
       "      <td>Responder</td>\n",
       "      <td>1.32</td>\n",
       "      <td>2.84</td>\n",
       "      <td>4.16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Patient-3</td>\n",
       "      <td>Patient-3-Tumor</td>\n",
       "      <td>Patient-3-Normal</td>\n",
       "      <td>Non-Responder</td>\n",
       "      <td>1.78</td>\n",
       "      <td>5.00</td>\n",
       "      <td>6.78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Patient-4</td>\n",
       "      <td>Patient-4-Tumor</td>\n",
       "      <td>Patient-4-Normal</td>\n",
       "      <td>Responder</td>\n",
       "      <td>4.93</td>\n",
       "      <td>10.50</td>\n",
       "      <td>15.43</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Patient_ID Tumor_Sample_Barcode Matched_Norm_Sample_Barcode       Response  \\\n",
       "0  Patient-0      Patient-0-Tumor            Patient-0-Normal  Non-Responder   \n",
       "1  Patient-1      Patient-1-Tumor            Patient-1-Normal      Responder   \n",
       "2  Patient-2      Patient-2-Tumor            Patient-2-Normal      Responder   \n",
       "3  Patient-3      Patient-3-Tumor            Patient-3-Normal  Non-Responder   \n",
       "4  Patient-4      Patient-4-Tumor            Patient-4-Normal      Responder   \n",
       "\n",
       "   Silent_mutations_per_Mb  Nonsynonymous_mutations_per_Mb  Mutations_per_Mb  \n",
       "0                     2.87                            6.77              9.64  \n",
       "1                     1.92                            6.14              8.06  \n",
       "2                     1.32                            2.84              4.16  \n",
       "3                     1.78                            5.00              6.78  \n",
       "4                     4.93                           10.50             15.43  "
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Preview sample-info.tsv file\n",
    "sample_info_path = os.path.join('vanallen-assessment', 'sample-information.tsv')\n",
    "sample_info_df = pd.read_csv(sample_info_path, sep='\\t')\n",
    "print(sample_info_df.columns)\n",
    "print('Number of patient entries:', len(sample_info_df))\n",
    "sample_info_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Hugo_Symbol', 'Chromosome', 'Start_position', 'End_position',\n",
      "       'Variant_Classification', 'Variant_Type', 'Reference_Allele',\n",
      "       'Tumor_Seq_Allele1', 'Tumor_Seq_Allele2', 'Tumor_Sample_Barcode',\n",
      "       'Matched_Norm_Sample_Barcode', 'Protein_Change', 't_alt_count',\n",
      "       't_ref_count'],\n",
      "      dtype='object')\n",
      "Number of annotation rows for Patient 0: 334\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_position</th>\n",
       "      <th>End_position</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele1</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Matched_Norm_Sample_Barcode</th>\n",
       "      <th>Protein_Change</th>\n",
       "      <th>t_alt_count</th>\n",
       "      <th>t_ref_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AMOT</td>\n",
       "      <td>X</td>\n",
       "      <td>112035152</td>\n",
       "      <td>112035152</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Patient-0-Tumor</td>\n",
       "      <td>Patient-0-Normal</td>\n",
       "      <td>p.Q612E</td>\n",
       "      <td>8</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SEMA6D</td>\n",
       "      <td>15</td>\n",
       "      <td>48062786</td>\n",
       "      <td>48062786</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Patient-0-Tumor</td>\n",
       "      <td>Patient-0-Normal</td>\n",
       "      <td>p.A676T</td>\n",
       "      <td>4</td>\n",
       "      <td>87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PRR12</td>\n",
       "      <td>19</td>\n",
       "      <td>50100969</td>\n",
       "      <td>50100969</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Patient-0-Tumor</td>\n",
       "      <td>Patient-0-Normal</td>\n",
       "      <td>p.S1126C</td>\n",
       "      <td>5</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TNR</td>\n",
       "      <td>1</td>\n",
       "      <td>175372529</td>\n",
       "      <td>175372529</td>\n",
       "      <td>Silent</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Patient-0-Tumor</td>\n",
       "      <td>Patient-0-Normal</td>\n",
       "      <td>p.S241S</td>\n",
       "      <td>30</td>\n",
       "      <td>155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CPA4</td>\n",
       "      <td>7</td>\n",
       "      <td>129944344</td>\n",
       "      <td>129944344</td>\n",
       "      <td>Silent</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Patient-0-Tumor</td>\n",
       "      <td>Patient-0-Normal</td>\n",
       "      <td>p.A137A</td>\n",
       "      <td>12</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Hugo_Symbol Chromosome  Start_position  End_position Variant_Classification  \\\n",
       "0        AMOT          X       112035152     112035152      Missense_Mutation   \n",
       "1      SEMA6D         15        48062786      48062786      Missense_Mutation   \n",
       "2       PRR12         19        50100969      50100969      Missense_Mutation   \n",
       "3         TNR          1       175372529     175372529                 Silent   \n",
       "4        CPA4          7       129944344     129944344                 Silent   \n",
       "\n",
       "  Variant_Type Reference_Allele Tumor_Seq_Allele1 Tumor_Seq_Allele2  \\\n",
       "0          SNP                G                 G                 C   \n",
       "1          SNP                G                 G                 A   \n",
       "2          SNP                C                 C                 G   \n",
       "3          SNP                G                 G                 A   \n",
       "4          SNP                C                 C                 G   \n",
       "\n",
       "  Tumor_Sample_Barcode Matched_Norm_Sample_Barcode Protein_Change  \\\n",
       "0      Patient-0-Tumor            Patient-0-Normal        p.Q612E   \n",
       "1      Patient-0-Tumor            Patient-0-Normal        p.A676T   \n",
       "2      Patient-0-Tumor            Patient-0-Normal       p.S1126C   \n",
       "3      Patient-0-Tumor            Patient-0-Normal        p.S241S   \n",
       "4      Patient-0-Tumor            Patient-0-Normal        p.A137A   \n",
       "\n",
       "   t_alt_count  t_ref_count  \n",
       "0            8           31  \n",
       "1            4           87  \n",
       "2            5           25  \n",
       "3           30          155  \n",
       "4           12          100  "
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Preview Patient 0 mutation .maf file\n",
    "patient0_path = os.path.join('vanallen-assessment', 'mafs', 'Patient-0.somatic.snvs.maf')\n",
    "patient0_df = pd.read_csv(patient0_path, sep='\\t')\n",
    "print(patient0_df.columns)\n",
    "print('Number of annotation rows for Patient 0:', len(patient0_df))\n",
    "patient0_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of annotation rows for all patients: 15673\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_position</th>\n",
       "      <th>End_position</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele1</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Matched_Norm_Sample_Barcode</th>\n",
       "      <th>Protein_Change</th>\n",
       "      <th>t_alt_count</th>\n",
       "      <th>t_ref_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CEP350</td>\n",
       "      <td>1</td>\n",
       "      <td>180063656</td>\n",
       "      <td>180063656</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Patient-36-Tumor</td>\n",
       "      <td>Patient-36-Normal</td>\n",
       "      <td>p.E2806K</td>\n",
       "      <td>12</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CCDC88C</td>\n",
       "      <td>14</td>\n",
       "      <td>91739009</td>\n",
       "      <td>91739009</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Patient-36-Tumor</td>\n",
       "      <td>Patient-36-Normal</td>\n",
       "      <td>p.G2016E</td>\n",
       "      <td>36</td>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KDM6B</td>\n",
       "      <td>17</td>\n",
       "      <td>7749509</td>\n",
       "      <td>7749509</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Patient-36-Tumor</td>\n",
       "      <td>Patient-36-Normal</td>\n",
       "      <td>p.Y117F</td>\n",
       "      <td>4</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PGAP1</td>\n",
       "      <td>2</td>\n",
       "      <td>197781268</td>\n",
       "      <td>197781268</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Patient-36-Tumor</td>\n",
       "      <td>Patient-36-Normal</td>\n",
       "      <td>p.F117L</td>\n",
       "      <td>11</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PARD6B</td>\n",
       "      <td>20</td>\n",
       "      <td>49366765</td>\n",
       "      <td>49366765</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Patient-36-Tumor</td>\n",
       "      <td>Patient-36-Normal</td>\n",
       "      <td>p.E287Q</td>\n",
       "      <td>21</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Hugo_Symbol Chromosome  Start_position  End_position Variant_Classification  \\\n",
       "0      CEP350          1       180063656     180063656      Missense_Mutation   \n",
       "1     CCDC88C         14        91739009      91739009      Missense_Mutation   \n",
       "2       KDM6B         17         7749509       7749509      Missense_Mutation   \n",
       "3       PGAP1          2       197781268     197781268      Missense_Mutation   \n",
       "4      PARD6B         20        49366765      49366765      Missense_Mutation   \n",
       "\n",
       "  Variant_Type Reference_Allele Tumor_Seq_Allele1 Tumor_Seq_Allele2  \\\n",
       "0          SNP                G                 G                 A   \n",
       "1          SNP                C                 C                 T   \n",
       "2          SNP                A                 A                 T   \n",
       "3          SNP                G                 G                 C   \n",
       "4          SNP                G                 G                 C   \n",
       "\n",
       "  Tumor_Sample_Barcode Matched_Norm_Sample_Barcode Protein_Change  \\\n",
       "0     Patient-36-Tumor           Patient-36-Normal       p.E2806K   \n",
       "1     Patient-36-Tumor           Patient-36-Normal       p.G2016E   \n",
       "2     Patient-36-Tumor           Patient-36-Normal        p.Y117F   \n",
       "3     Patient-36-Tumor           Patient-36-Normal        p.F117L   \n",
       "4     Patient-36-Tumor           Patient-36-Normal        p.E287Q   \n",
       "\n",
       "   t_alt_count  t_ref_count  \n",
       "0           12           28  \n",
       "1           36           57  \n",
       "2            4           25  \n",
       "3           11           25  \n",
       "4           21           88  "
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load in all patient .maf data as single dataframe\n",
    "maf_dir = os.listdir(os.path.join('vanallen-assessment', 'mafs'))  # patient .maf data folder\n",
    "mutations_df = pd.DataFrame()  # blank DF for later concatenation\n",
    "for item in maf_dir:\n",
    "    patient_df = pd.read_csv(os.path.join('vanallen-assessment', 'mafs', item), sep='\\t')\n",
    "    mutations_df = pd.concat([mutations_df, patient_df], ignore_index=True)\n",
    "print('Number of annotation rows for all patients:', len(mutations_df))\n",
    "mutations_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of entries in merged_df: 15673\n",
      "Mutation variant types: ['Missense_Mutation' 'Silent' 'Nonsense_Mutation' 'Splice_Site']\n",
      "Number of nonsynonymous mutations: 11247\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_position</th>\n",
       "      <th>End_position</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele1</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Matched_Norm_Sample_Barcode</th>\n",
       "      <th>Protein_Change</th>\n",
       "      <th>t_alt_count</th>\n",
       "      <th>t_ref_count</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Response</th>\n",
       "      <th>Silent_mutations_per_Mb</th>\n",
       "      <th>Nonsynonymous_mutations_per_Mb</th>\n",
       "      <th>Mutations_per_Mb</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CEP350</td>\n",
       "      <td>1</td>\n",
       "      <td>180063656</td>\n",
       "      <td>180063656</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Patient-36-Tumor</td>\n",
       "      <td>Patient-36-Normal</td>\n",
       "      <td>p.E2806K</td>\n",
       "      <td>12</td>\n",
       "      <td>28</td>\n",
       "      <td>Patient-36</td>\n",
       "      <td>Responder</td>\n",
       "      <td>2.47</td>\n",
       "      <td>6.0</td>\n",
       "      <td>8.47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CCDC88C</td>\n",
       "      <td>14</td>\n",
       "      <td>91739009</td>\n",
       "      <td>91739009</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Patient-36-Tumor</td>\n",
       "      <td>Patient-36-Normal</td>\n",
       "      <td>p.G2016E</td>\n",
       "      <td>36</td>\n",
       "      <td>57</td>\n",
       "      <td>Patient-36</td>\n",
       "      <td>Responder</td>\n",
       "      <td>2.47</td>\n",
       "      <td>6.0</td>\n",
       "      <td>8.47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KDM6B</td>\n",
       "      <td>17</td>\n",
       "      <td>7749509</td>\n",
       "      <td>7749509</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Patient-36-Tumor</td>\n",
       "      <td>Patient-36-Normal</td>\n",
       "      <td>p.Y117F</td>\n",
       "      <td>4</td>\n",
       "      <td>25</td>\n",
       "      <td>Patient-36</td>\n",
       "      <td>Responder</td>\n",
       "      <td>2.47</td>\n",
       "      <td>6.0</td>\n",
       "      <td>8.47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PGAP1</td>\n",
       "      <td>2</td>\n",
       "      <td>197781268</td>\n",
       "      <td>197781268</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Patient-36-Tumor</td>\n",
       "      <td>Patient-36-Normal</td>\n",
       "      <td>p.F117L</td>\n",
       "      <td>11</td>\n",
       "      <td>25</td>\n",
       "      <td>Patient-36</td>\n",
       "      <td>Responder</td>\n",
       "      <td>2.47</td>\n",
       "      <td>6.0</td>\n",
       "      <td>8.47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PARD6B</td>\n",
       "      <td>20</td>\n",
       "      <td>49366765</td>\n",
       "      <td>49366765</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Patient-36-Tumor</td>\n",
       "      <td>Patient-36-Normal</td>\n",
       "      <td>p.E287Q</td>\n",
       "      <td>21</td>\n",
       "      <td>88</td>\n",
       "      <td>Patient-36</td>\n",
       "      <td>Responder</td>\n",
       "      <td>2.47</td>\n",
       "      <td>6.0</td>\n",
       "      <td>8.47</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Hugo_Symbol Chromosome  Start_position  End_position Variant_Classification  \\\n",
       "0      CEP350          1       180063656     180063656      Missense_Mutation   \n",
       "1     CCDC88C         14        91739009      91739009      Missense_Mutation   \n",
       "2       KDM6B         17         7749509       7749509      Missense_Mutation   \n",
       "3       PGAP1          2       197781268     197781268      Missense_Mutation   \n",
       "4      PARD6B         20        49366765      49366765      Missense_Mutation   \n",
       "\n",
       "  Variant_Type Reference_Allele Tumor_Seq_Allele1 Tumor_Seq_Allele2  \\\n",
       "0          SNP                G                 G                 A   \n",
       "1          SNP                C                 C                 T   \n",
       "2          SNP                A                 A                 T   \n",
       "3          SNP                G                 G                 C   \n",
       "4          SNP                G                 G                 C   \n",
       "\n",
       "  Tumor_Sample_Barcode Matched_Norm_Sample_Barcode Protein_Change  \\\n",
       "0     Patient-36-Tumor           Patient-36-Normal       p.E2806K   \n",
       "1     Patient-36-Tumor           Patient-36-Normal       p.G2016E   \n",
       "2     Patient-36-Tumor           Patient-36-Normal        p.Y117F   \n",
       "3     Patient-36-Tumor           Patient-36-Normal        p.F117L   \n",
       "4     Patient-36-Tumor           Patient-36-Normal        p.E287Q   \n",
       "\n",
       "   t_alt_count  t_ref_count  Patient_ID   Response  Silent_mutations_per_Mb  \\\n",
       "0           12           28  Patient-36  Responder                     2.47   \n",
       "1           36           57  Patient-36  Responder                     2.47   \n",
       "2            4           25  Patient-36  Responder                     2.47   \n",
       "3           11           25  Patient-36  Responder                     2.47   \n",
       "4           21           88  Patient-36  Responder                     2.47   \n",
       "\n",
       "   Nonsynonymous_mutations_per_Mb  Mutations_per_Mb  \n",
       "0                             6.0              8.47  \n",
       "1                             6.0              8.47  \n",
       "2                             6.0              8.47  \n",
       "3                             6.0              8.47  \n",
       "4                             6.0              8.47  "
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge mutations_df and sample_info_df\n",
    "merged_df = mutations_df.merge(sample_info_df, how='left', on=['Tumor_Sample_Barcode', 'Matched_Norm_Sample_Barcode'])\n",
    "print('Number of entries in merged_df:', len(merged_df))\n",
    "\n",
    "# Filter out silent mutations from variant_classification column\n",
    "print('Mutation variant types:', merged_df['Variant_Classification'].unique())\n",
    "nonsynon_df = merged_df[merged_df['Variant_Classification'] != 'Silent']  # nonsynonymous mutations df\n",
    "print('Number of nonsynonymous mutations:', len(nonsynon_df))\n",
    "nonsynon_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_position</th>\n",
       "      <th>End_position</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele1</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Matched_Norm_Sample_Barcode</th>\n",
       "      <th>Protein_Change</th>\n",
       "      <th>t_alt_count</th>\n",
       "      <th>t_ref_count</th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Response</th>\n",
       "      <th>Silent_mutations_per_Mb</th>\n",
       "      <th>Nonsynonymous_mutations_per_Mb</th>\n",
       "      <th>Mutations_per_Mb</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>STXBP3</td>\n",
       "      <td>1</td>\n",
       "      <td>109294878</td>\n",
       "      <td>109294878</td>\n",
       "      <td>Splice_Site</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Patient-36-Tumor</td>\n",
       "      <td>Patient-36-Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32</td>\n",
       "      <td>61</td>\n",
       "      <td>Patient-36</td>\n",
       "      <td>Responder</td>\n",
       "      <td>2.47</td>\n",
       "      <td>6.0</td>\n",
       "      <td>8.47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>SFXN5</td>\n",
       "      <td>2</td>\n",
       "      <td>73198699</td>\n",
       "      <td>73198699</td>\n",
       "      <td>Splice_Site</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Patient-36-Tumor</td>\n",
       "      <td>Patient-36-Normal</td>\n",
       "      <td>p.H247H</td>\n",
       "      <td>8</td>\n",
       "      <td>12</td>\n",
       "      <td>Patient-36</td>\n",
       "      <td>Responder</td>\n",
       "      <td>2.47</td>\n",
       "      <td>6.0</td>\n",
       "      <td>8.47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>SPG7</td>\n",
       "      <td>16</td>\n",
       "      <td>89590413</td>\n",
       "      <td>89590413</td>\n",
       "      <td>Splice_Site</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Patient-36-Tumor</td>\n",
       "      <td>Patient-36-Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>Patient-36</td>\n",
       "      <td>Responder</td>\n",
       "      <td>2.47</td>\n",
       "      <td>6.0</td>\n",
       "      <td>8.47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>NEO1</td>\n",
       "      <td>15</td>\n",
       "      <td>73590688</td>\n",
       "      <td>73590688</td>\n",
       "      <td>Splice_Site</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Patient-36-Tumor</td>\n",
       "      <td>Patient-36-Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "      <td>16</td>\n",
       "      <td>Patient-36</td>\n",
       "      <td>Responder</td>\n",
       "      <td>2.47</td>\n",
       "      <td>6.0</td>\n",
       "      <td>8.47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>ABCA7</td>\n",
       "      <td>19</td>\n",
       "      <td>1057312</td>\n",
       "      <td>1057312</td>\n",
       "      <td>Splice_Site</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Patient-36-Tumor</td>\n",
       "      <td>Patient-36-Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32</td>\n",
       "      <td>66</td>\n",
       "      <td>Patient-36</td>\n",
       "      <td>Responder</td>\n",
       "      <td>2.47</td>\n",
       "      <td>6.0</td>\n",
       "      <td>8.47</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Hugo_Symbol Chromosome  Start_position  End_position  \\\n",
       "27       STXBP3          1       109294878     109294878   \n",
       "35        SFXN5          2        73198699      73198699   \n",
       "65         SPG7         16        89590413      89590413   \n",
       "128        NEO1         15        73590688      73590688   \n",
       "160       ABCA7         19         1057312       1057312   \n",
       "\n",
       "    Variant_Classification Variant_Type Reference_Allele Tumor_Seq_Allele1  \\\n",
       "27             Splice_Site          SNP                G                 G   \n",
       "35             Splice_Site          SNP                G                 G   \n",
       "65             Splice_Site          SNP                G                 G   \n",
       "128            Splice_Site          SNP                G                 G   \n",
       "160            Splice_Site          SNP                G                 G   \n",
       "\n",
       "    Tumor_Seq_Allele2 Tumor_Sample_Barcode Matched_Norm_Sample_Barcode  \\\n",
       "27                  C     Patient-36-Tumor           Patient-36-Normal   \n",
       "35                  A     Patient-36-Tumor           Patient-36-Normal   \n",
       "65                  A     Patient-36-Tumor           Patient-36-Normal   \n",
       "128                 C     Patient-36-Tumor           Patient-36-Normal   \n",
       "160                 C     Patient-36-Tumor           Patient-36-Normal   \n",
       "\n",
       "    Protein_Change  t_alt_count  t_ref_count  Patient_ID   Response  \\\n",
       "27             NaN           32           61  Patient-36  Responder   \n",
       "35         p.H247H            8           12  Patient-36  Responder   \n",
       "65             NaN            5           10  Patient-36  Responder   \n",
       "128            NaN           11           16  Patient-36  Responder   \n",
       "160            NaN           32           66  Patient-36  Responder   \n",
       "\n",
       "     Silent_mutations_per_Mb  Nonsynonymous_mutations_per_Mb  Mutations_per_Mb  \n",
       "27                      2.47                             6.0              8.47  \n",
       "35                      2.47                             6.0              8.47  \n",
       "65                      2.47                             6.0              8.47  \n",
       "128                     2.47                             6.0              8.47  \n",
       "160                     2.47                             6.0              8.47  "
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nonsynon_df[nonsynon_df['Variant_Classification']=='Splice_Site'].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Missense_Mutation False\n",
      "Nonsense_Mutation False\n",
      "Splice_Site True\n",
      "Number of splice_site mutations: 392\n",
      "Number of splice_site NaN values: 209\n",
      "With NaN Variant_Classification values:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Protein_Change</th>\n",
       "      <th>Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ERBB4</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.S1289A</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ERBB3</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.H228Q</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ERBB4</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.Q707E</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.E545K</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TYRO3</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.L819M</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ERBB3</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.M91I</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ERBB4</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.E317K</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>RXRA</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.S330F</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.F53L</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>FAM47C</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.Q225E</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>KRTAP4-11</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.L161V</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TP53</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.R248Q</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>PLXNA2</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.E1480K</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>KRTAP4-4</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>p.H62R</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SPTAN1</td>\n",
       "      <td>Splice_Site</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Hugo_Symbol Variant_Classification Protein_Change  Count\n",
       "0        ERBB4      Missense_Mutation       p.S1289A     14\n",
       "1        ERBB3      Missense_Mutation        p.H228Q      5\n",
       "2        ERBB4      Missense_Mutation        p.Q707E      5\n",
       "3       PIK3CA      Missense_Mutation        p.E545K      5\n",
       "4        TYRO3      Missense_Mutation        p.L819M      4\n",
       "5        ERBB3      Missense_Mutation         p.M91I      4\n",
       "6        ERBB4      Missense_Mutation        p.E317K      4\n",
       "7         RXRA      Missense_Mutation        p.S330F      4\n",
       "8       MAP2K1      Missense_Mutation         p.F53L      4\n",
       "9       FAM47C      Missense_Mutation        p.Q225E      4\n",
       "10   KRTAP4-11      Missense_Mutation        p.L161V      4\n",
       "11        TP53      Missense_Mutation        p.R248Q      4\n",
       "12      PLXNA2      Missense_Mutation       p.E1480K      3\n",
       "13    KRTAP4-4      Missense_Mutation         p.H62R      3\n",
       "14      SPTAN1            Splice_Site            NaN      3"
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Find the 15 most common mutations\n",
    "### NOTE: accounting for NaN values unique to splice_site variants yields different results\n",
    "for variant in nonsynon_df['Variant_Classification'].unique():\n",
    "    nan_check = nonsynon_df[nonsynon_df['Variant_Classification']==variant].isnull().values.any()\n",
    "    print(variant, nan_check)  # check for NaN values in variant_classifications\n",
    "print('Number of splice_site mutations:', len(nonsynon_df[nonsynon_df['Variant_Classification']=='Splice_Site']))  # 392 splice_sites\n",
    "print('Number of splice_site NaN values:', nonsynon_df['Protein_Change'].isnull().sum())  # 209 splice_site NaNs\n",
    "\n",
    "# Not accounting for splice_site variant classification (drop NaN protein_change values)\n",
    "# top15_noNan = nonsynon_df[['Hugo_Symbol', 'Variant_Classification', 'Protein_Change']].value_counts().to_frame('Count').reset_index()\n",
    "# print('Without NaN Variant_Classification values: \\n', top15_noNan.head(15))\n",
    "\n",
    "# Accounting for splice_site variant classification (don't drop NaN protein_change values)\n",
    "top15_wSplice = nonsynon_df[['Hugo_Symbol', 'Variant_Classification', 'Protein_Change']].value_counts(dropna=False).to_frame('Count').reset_index()\n",
    "print('With NaN Variant_Classification values:')\n",
    "top15_wSplice.head(15)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  },
  "vscode": {
   "interpreter": {
    "hash": "2a8770c472293afed95e339a9700fa30752e899dfa15f80b3a28dc1457ff229a"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
